topo_processor/util/aws_credentials.py

```diff
@@ -1,29 +1,55 @@
+import json
 import os
+from typing import Dict
 
 import boto3
 
-default_credentials = {}
-session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
-client = session.client("sts")
-bucket_roles = {}
+from topo_processor.file_system.get_fs import bucket_name_from_path
+from topo_processor.util.configuration import aws_role_config_path
+
+
+class Credentials:
+    access_key: str
+    secret_key: str
+    token: str
+
+    def __init__(self, access_key: str, secret_key: str, token: str):
+        self.access_key = access_key
+        self.secret_key = secret_key
+        self.token = token
 
 
-def get_credentials(bucket_name: str):
+default_credentials: Credentials
+session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
+client: boto3.client = session.client("sts")
+bucket_roles: Dict[str, str] = {}
+assume_role_cache: Dict = {}
+
 
+def get_credentials(bucket_name: str) -> Credentials:
```

Comment:
these credentials need to be refreshed somewhere, was that taken out of this PR?

Reply:
I've already spent a lot of more time that I've expected on this. I would prefer do that in another PR. I've created https://linzgovt.atlassian.net/browse/TDE-235